mysql-test/suite/multi_source/connects_tried.test

```diff
@@ -0,0 +1,133 @@
+# MDEV-35304: Test the `Connects_Tried` feature
+#
+# Two connections with different retry frequencies tests their
+# separate counters in parallel multi-source (SHOW SLAVE [name] STATUS).
+#
+# Note: nearly all SELECT results are timing-sensitive,
+# they therefore list unexpected rows because.
+
+--source include/have_binlog_format_mixed.inc # The test is agnostic of binlog formats.
+# `rpl_*.inc` (still) expects `server_1` be `master` and `server_2` be `slave`.
+--let $rpl_server_count= 3
+--let $rpl_skip_start_slave= 1
+--source include/master-slave.inc
```

Comment:
Conventionally, our tests in `multi_source` don't actually use `master-slave.inc` or `rpl_init.inc`, but rather create the connections manually so each channel/source is clear in its use in the test.  

Note too (for the `rpl` suite) when creating tests cases with topologies beyond the simple master-slave, you can use `rpl_init.inc` (instead of `master-slave.inc`) with the variable `rpl_topology=1->3,2->3`, etc, and it will set everything up for you.

Reply:
My test used the convenience include `rpl_start_server.inc`, which `--source`s `rpl_reconnect.inc`, whose code is designed to explicitly (and consequentially *only*) work with  `master-slave.inc` 😕.
I’m not certain how to lift that limitation (#3849), so I decided to stick to the `rpl` format (which isn’t the `multi_source` format ¯\\\_(ツ)_/¯) over writing the de/reconnection commands out.
I saw that our test engineer’s got plans on generalizing setups for QA, which could alleviate.

> `rpl_topology=1->3,2->3`

I thought `rpl_init.inc` only executes unnamed `CHANGE MASTER TO`, which `2->3` would overwrite `1->3` (overwrite `server_3`’s primary from port 1 to port 2).